### PR TITLE
Don't need minitest nor mocha to test srb init

### PR DESCRIPTION
**Pull request by jez**
Turns out that `srb init` explicitly doesn't work in projects without Gemfiles,
so I deleted the part of this test where I thought that was supposed to work
but in fact it was only working because it was using the `Gemfile` from the
parent directory.

### Motivation

Improves #1369.

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by jez** [comment]
note: it works:

https://buildkite.com/sorbet/sorbet/builds/4645#369ce593-9f79-4160-94b5-d1166e0d9e9a/67-189

i plan to remove the testing changes so that this goes back to only building on master.

File: .buildkite/build-static-release.sh [hunk 1]
```diff
@@ -74,14 +75,17 @@ else
...
+trap "rm -rf '$smoke_test_dir'" EXIT
 
-  rm -rf sorbet/
+cp ../../.ruby-version "$smoke_test_dir"
 
+(
+  # Make sure we're in a totally separate tree, otherwise a Gemfile from a
+  # parent folder will affect this test.
+  cd "$smoke_test_dir"
   touch Gemfile
```
Comment (elliottt) [inline .buildkite/build-static-release.sh]: Do we want to support using sorbet outside of a bundler project?

Comment (jez) [reply .buildkite/build-static-release.sh]: Every phase of `srb` needs to know what gems you're using. If you don't have a Gemfile that declares them, we'd have to like search your gem path to find what you *could* potentially require. I think that `srb tc` should still work regardless.

But at this point i think bundler is so ubiquitous that it's not a high priority.

Comment (jez) [reply .buildkite/build-static-release.sh]: That should also be separate from improving this test. I'm happy to have an issue to track it, but i would kind of prefer to have someone ask for it first so we can really understand what they're asking for.

Comment (elliottt) [reply .buildkite/build-static-release.sh]: That makes sense, thanks!

**Pull request merged**